src/Commands/IndexCommand.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Rapidez\Core\Commands;
+
+use Illuminate\Console\Command;
+use Rapidez\Core\Events\IndexAfterEvent;
+use Rapidez\Core\Events\IndexBeforeEvent;
+use Rapidez\Core\Facades\Rapidez;
+use Rapidez\Core\Models\Traits\Searchable;
+
+class IndexCommand extends Command
+{
+    protected $signature = 'rapidez:index {--t|types= : To specify types of models to index, separated by commas} {--s|store= : To specify store IDs from Magento, separated by commas}';
+
+    protected $description = 'Index all searchable models into ElasticSearch';
+
+    public function handle()
+    {
+        $baseSearchableModels = collect(config('rapidez.models'))
+            ->filter(fn ($class) => in_array(Searchable::class, class_uses_recursive($class)))
+            ->merge(config('rapidez.indexer.extra_models'));
+
+        $types = $this->option('types')
+            ? $baseSearchableModels->only(explode(',', $this->option('types')))
+            : $baseSearchableModels;
+
+        $stores = $this->option('store')
+            ? Rapidez::getStores(explode(',', $this->option('store')))
+            : Rapidez::getStores();
+
+        $this->call('cache:clear');
+
+        IndexBeforeEvent::dispatch($this);
+
+        foreach ($stores as $store) {
+            Rapidez::setStore($store);
+
+            $this->line('Store: ' . $store['name']);
+
+            foreach ($types as $type => $model) {
+                config()->set('rapidez.index', $type);
```

Comment:
Is it possible to infer this from the model itself? It would save us from setting this value which thanks to your changes doesn't seem needed anymore.

Perhaps a function that reads from an empty model variable/constant, which if it's not set [Str::snake()](https://laravel.com/docs/master/strings#method-snake-case) on the class name

Reply:
That's also an option; currently we do this differently on multiple places, for example:

- The category index is defined here: https://github.com/rapidez/core/pull/735/files#diff-694b39ded25e81aa9cfa9de2b404533c0f924a4919b0f68b38aebfb0741ff68fR1 which is using this config: https://github.com/rapidez/core/pull/735/files#diff-fe38fec1955243f9e20c1cdc08b64167096d11ee224c3c86fc8e4f4671750a51R64
- For the products we've this: https://github.com/rapidez/core/pull/706/files#diff-b30cee12bf5e6816f22522e4ab112a4dc162789811fdcd04c7df321dcb438dbbR132 which can be used in the frontend with `config.index`